### PR TITLE
doc: atlas-basic README and LICENSE files

### PR DESCRIPTION
**Pull request by rubenVB01**
## Description

This PR contains the global README and LICENSE files for the atlas-basic terraform module. 

## Considerations

There are some links that still need to be updated, but can't in this PR because the module needs to be published in the Terraform Public Registry. In Ticket CLOUDP-262997 this will be done.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+This Terraform module creates a basic setup on [MongoDB Atlas] [atlas]. 
+
+It creates the following resources:
+
+- A MongoDB Atlas project.
+- A MongoDB Atlas cluster provisioned with AWS, GCP or AZURE provider. It can also be a free tier (TENANT) cluster.
+- Zero or more IP addresses.
+- Zero or more CIDR blocks.
+- One or more database users.
```
Comment (JuliaMongo) [inline README.md:11]: I think the sequence is:
project, then users, then IP addresses, then CIDR blocks then cluster

File: README.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+You can find detailed information of the module's input and output variables in the Terraform Public Registry <LINK_TO_REGISTRY>
```
Comment (JuliaMongo) [inline README.md:13]: This is a missing link. Should it be this way?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  electable_specs = {
+	instance_size = "M10"
+  }
+  analytics_specs = {
+	instance_size = "M10"	
+  }
+}
+```
+
+The [examples] [Ex] folder contains detailed examples that shows how to use this module.
```
Comment (JuliaMongo) [inline README.md:55]: that show, not "that shows"

File: README.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+## Cluster Creation Considerations 
+
+The `atlas-basic` module only supports the Replica Set cluster type. This setup is straightforward to manage and involves fewer components and configuration steps, making it ideal for users who are new to MongoDB or those needing a quick and simple deployment. Additionally, it is the most common type of cluster.
```
Comment (JuliaMongo) [inline README.md:59]: replica set is the most common type of cluster

File: README.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+MongoDB Atlas clusters support several cloud service providers for server provisioning. The possible values are: 
+
+- Amazon AWS (default for this module)
+- Google Cloud Platform
+- Microsoft Azure
+- Multi-tenant cluster (free tier cluster)
+
+### Considerations and Limitations
+
+- If specifying a provider other than the default, you must also specify the region name, as not all providers support the same regions. See the reference list for [AWS] [aws-region], [GCP] [gcp-region], [AZURE] [azure-region].
```
Comment (JuliaMongo) [inline README.md:72]: Is there a default cloud provider? I didn't know there is one.

Comment (JuliaMongo) [reply README.md:72]: Also do you have links???

File: README.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+- Microsoft Azure
+- Multi-tenant cluster (free tier cluster)
+
+### Considerations and Limitations
+
+- If specifying a provider other than the default, you must also specify the region name, as not all providers support the same regions. See the reference list for [AWS] [aws-region], [GCP] [gcp-region], [AZURE] [azure-region]. 
+
+- For TENANT clusters:
+    - Analytics nodes cannot be defined due to resource limitations and cost management considerations.
+    - The instance size of the electable nodes must be either M0, M2 or M5, which are the free tier or low-cost tier MongoDB Atlas clusters.
```
Comment (JuliaMongo) [inline README.md:76]: I am pretty sure the free tier now is more than m5 too, please check this again.

File: README.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+- If specifying a provider other than the default, you must also specify the region name, as not all providers support the same regions. See the reference list for [AWS] [aws-region], [GCP] [gcp-region], [AZURE] [azure-region]. 
+
+- For TENANT clusters:
+    - Analytics nodes cannot be defined due to resource limitations and cost management considerations.
+    - The instance size of the electable nodes must be either M0, M2 or M5, which are the free tier or low-cost tier MongoDB Atlas clusters.  
+
+## Authors
+
+Module is maintained by the MongoDB API Experience Integrations team.
```
Comment (JuliaMongo) [inline README.md:80]: Who is that and do we have a link?

This module is maintained

**Review by JuliaMongo — APPROVED** [review]
With comments.

File: README.md (hunk 2 above)
Comment (rubenVB01) [reply README.md:13]: For now yes, once we publish the module on the public registry we will have the correct link.

File: README.md (hunk 5 above)
Comment (rubenVB01) [reply README.md:72]: We are choosing the default (in this case, we chose AWS). The links are at the bottom of the .md

File: README.md [hunk 8]
```diff
@@ -0,0 +1,89 @@
...
+
+This Terraform module creates a basic setup on [MongoDB Atlas] [atlas]. 
+
+It creates the following resources:
+
+- A MongoDB Atlas project.
+- One or more database users.
+- Zero or more IP addresses.
+- Zero or more CIDR blocks.
+- A MongoDB Atlas cluster provisioned with AWS, GCP or AZURE provider. It can also be a free tier (TENANT) cluster.
```
Comment (AgustinBettati) [inline README.md:11]: should we be explicit that it is a replica set?

File: README.md [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+### Provider
+
+MongoDB Atlas clusters support several cloud service providers for server provisioning. The possible values are: 
+
+- Amazon AWS (the one we chose as default for this module)
+- Google Cloud Platform
+- Microsoft Azure
+- Multi-tenant cluster (free tier cluster)
+
+### Considerations and Limitations
```
Comment (AgustinBettati) [inline README.md:70]: After seeing this again believe this sub title does not add much value, maybe leave this content within the provider section?

File: README.md [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+}
+```
+
+The [examples] [Ex] folder contains detailed examples that show how to use this module.
+
+## Cluster Creation Considerations 
+
+The `atlas-basic` module only supports the Replica Set cluster type. This setup is straightforward to manage and involves fewer components and configuration steps, making it ideal for users who are new to MongoDB or those needing a quick and simple deployment. Additionally, Replica Set is the most common type of cluster.
+
+### Provider
```
Comment (AgustinBettati) [inline README.md:61]:
```suggestion
### Supported Cloud Providers
```

**Review by AgustinBettati — APPROVED** [review]
LGTM, leaving one suggestion

File: README.md (hunk 8 above)
Comment (rubenVB01) [reply README.md:11]: It is later explained in the considerations section, I'm not sure if it would be too redundant

**Pull request merged**